### PR TITLE
Add upper bounds for menhir 20211215

### PR DESCRIPTION
**Pull request by mseri**
Needed before merging https://github.com/ocaml/opam-repository/pull/20256

Once the upper bounds are correct, we can merge this and re-test the other PR

Signed-off-by: Marcello Seri <marcello.seri@gmail.com>

### CODE REVIEW

**Comment by mseri** [comment]
Notes:
- `pds` seems to be failing due to `toml` failing
- some versions of `zipperposition` were failing due to `logtk` failing

The changes here reflect that.

Please double check that they are correct

File: packages/atd/atd.1.13.0/opam [hunk 1]
```diff
@@ -13,7 +13,7 @@ build: [
 depends: [
   "ocaml" {>= "4.02.3"}
   "jbuilder"
-  "menhir" {build}
+  "menhir" {nuild & < "20211215"}
```
Comment (fpottier) [inline packages/atd/atd.1.13.0/opam:16]: nuild should be build!

Comment (mseri) [reply packages/atd/atd.1.13.0/opam:16]:
```suggestion
  "menhir" {build & < "20211215"}
```

Comment (mseri) [reply packages/atd/atd.1.13.0/opam]: Thanks!

**Comment by mseri** [comment]
`atdgen 2.2.1` was getting `Re` transitively: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/2ab24cbeed476172e94d7d5f319705628cb2b0d1/variant/opam-2.0,compilers,4.13,atd.2.0.0,revdeps,atdgen.2.2.1

Not sure what upper bound it needs, I don't think we should fix it now:
```
#=== ERROR while compiling atdgen.2.2.1 =======================================#
# context              2.0.10 | linux/x86_64 | ocaml-base-compiler.4.13.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.13/.opam-switch/build/atdgen.2.2.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p atdgen -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/atdgen-1193-e7966f.env
# output-file          ~/.opam/log/atdgen-1193-e7966f.out
### output ###
#     ocamlopt atdgen/src/.atdgen_emit.objs/native/atdgen_emit__Ocaml.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.13/bin/ocamlopt.opt -w -40 -w -27 -safe-string -g -I atdgen/src/.atdgen_emit.objs/byte -I atdgen/src/.atdgen_emit.objs/native -I /home/opam/.opam/4.13/lib/atd -I /home/opam/.opam/4.13/lib/biniou -I /home/opam/.opam/4.13/lib/easy-format -I /home/opam/.opam/4.13/lib/yojson -intf-suffix .ml -no-alias-deps -open Atdgen_emit -o atdgen/src/.atdgen_emit.objs/native/atdgen_emit__Ocaml.cmx -c -impl atdgen/src/ocaml.ml)
# File "atdgen/src/ocaml.ml", line 618, characters 12-24:
# 618 | let split = Re.Str.split (Re.Str.regexp " ")
#                   ^^^^^^^^^^^^
# Error: Unbound module Re
```

**Comment by mseri** [comment]
`coin` fails with (https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/2ab24cbeed476172e94d7d5f319705628cb2b0d1/variant/opam-2.1,compilers,4.08,coin.0.1.0)
```
#=== ERROR while compiling coin.0.1.0 =========================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/dinosaure/coin/releases/download/v0.1.0/coin-0.1.0.tbz)
# path                 ~/.opam/4.08/.opam-switch/build/coin.0.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build jbuilder build -p coin -j 31
# exit-code            1
# env-file             ~/.opam/log/coin-7323-6b6fc4.env
# output-file          ~/.opam/log/coin-7323-6b6fc4.out
### output ###
#       ocamlc gen/.generate.eobjs/generate.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I gen/.generate.eobjs -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bos -I /home/opam/.opam/4.08/lib/cmdliner -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/fpath -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/ptmap -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -no-alias-deps -o gen/.generate.eobjs/generate.cmo -c -impl gen/generate.ml)
# File "gen/generate.ml", line 12, characters 7-13:
# 12 |   Fmt.(suffix (const string "|]") @@ prefix (const string "[|") @@ list ~sep int) ppf lst
#             ^^^^^^
# Alert deprecated: Fmt.suffix
# use Fmt.(++) instead.
# File "gen/generate.ml", line 12, characters 37-43:
# 12 |   Fmt.(suffix (const string "|]") @@ prefix (const string "[|") @@ list ~sep int) ppf lst
#                                           ^^^^^^
# Alert deprecated: Fmt.prefix
# use Fmt.(++) instead.
# File "gen/generate.ml", line 41, characters 4-7:
# 41 | let cmd =
#          ^^^
# Error: The type of this expression,
#        ((unit, _[> Rresult.R.msg ]) Bos.OS.result, [> Rresult.R.msg ])
#        Rresult.result Cmdliner.Term.t * Cmdliner.Term.info,
#        contains type variables that cannot be generalized
```

I don't know the fix for this

**Comment by mseri** [comment]
same for `uuuu` 0.1.0 and 0.1.1 (https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/2ab24cbeed476172e94d7d5f319705628cb2b0d1/variant/opam-2.1,compilers,4.08,uuuu.0.1.1):
```

#=== ERROR while compiling uuuu.0.1.0 =========================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/dinosaure/uuuu/releases/download/v0.1.0/uuuu-0.1.0.tbz)
# path                 ~/.opam/4.08/.opam-switch/build/uuuu.0.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build jbuilder build -p uuuu -j 31
# exit-code            1
# env-file             ~/.opam/log/uuuu-7324-e587a0.env
# output-file          ~/.opam/log/uuuu-7324-e587a0.out
### output ###
#       ocamlc gen/.generate.eobjs/generate.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I gen/.generate.eobjs -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bos -I /home/opam/.opam/4.08/lib/cmdliner -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/fpath -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/ptmap -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -no-alias-deps -o gen/.generate.eobjs/generate.cmo -c -impl gen/generate.ml)
# File "gen/generate.ml", line 12, characters 7-13:
# 12 |   Fmt.(suffix (const string "|]") @@ prefix (const string "[|") @@ list ~sep int) ppf lst
#             ^^^^^^
# Alert deprecated: Fmt.suffix
# use Fmt.(++) instead.
# File "gen/generate.ml", line 12, characters 37-43:
# 12 |   Fmt.(suffix (const string "|]") @@ prefix (const string "[|") @@ list ~sep int) ppf lst
#                                           ^^^^^^
# Alert deprecated: Fmt.prefix
# use Fmt.(++) instead.
# File "gen/generate.ml", line 41, characters 4-7:
# 41 | let cmd =
#          ^^^
# Error: The type of this expression,
#        ((unit, _[> Rresult.R.msg ]) Bos.OS.result, [> Rresult.R.msg ])
#        Rresult.result Cmdliner.Term.t * Cmdliner.Term.info,
#        contains type variables that cannot be generalized
```

**Comment by mseri** [comment]
`atd2conv` 0.0.0 fails with
```
#=== ERROR while compiling atd2cconv.0.0.0 ====================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.13.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.13/.opam-switch/build/atd2cconv.0.0.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build make -f Makefile-ocamlbuild
# exit-code            2
# env-file             ~/.opam/log/atd2cconv-462-775d16.env
# output-file          ~/.opam/log/atd2cconv-462-775d16.out
### output ###
[...]
# ocamlfind ocamlopt -shared -linkall -thread -package atd,nonstd,smart_print -I src/lib src/lib/atd2cconv.cmxa -o src/lib/atd2cconv.cmxs
# ocamlfind ocamlc -thread -package atd,nonstd,smart_print -linkpkg \
#   -I _build/src/lib/ atd2cconv.cma src/app/main.ml -o atd2cconv
# File "src/app/main.ml", line 38, characters 30-34:
# 38 |   let sorted = Atd_util.tsort body in
#                                    ^^^^
# Error: This expression has type Atd_ast.annot = Atd_ast.annot_section list
#        but an expression was expected of type
#          Atd_ast.module_body = Atd_ast.module_item list
#        Type
#          Atd_ast.annot_section =
#            string * (Atd_ast.loc * Atd_ast.annot_field list)
#        is not compatible with type
#          Atd_ast.module_item = [ `Type of Atd_ast.type_def ] 
# make: *** [Makefile-ocamlbuild:7: byte] Error 2
```

I don't know the fix for this

**Comment by mseri** [comment]
Many issues have been fixed, I suggest to merge to be able to move on with menhir's PR

**Comment by kit-ty-kate** [comment]
This all looks fine. Thanks!

**Pull request merged**